test/Chainweb/Test/Pact/PactInProcApi.hs

```diff
@@ -296,128 +334,105 @@ newBlockRewindValidate mpRefIO reqIO = testCase "newBlockRewindValidate" $ do
 
     chainDataMemPool = mempty {
       mpaGetBlock = \_ _ _ _ bh -> do
-          fmap V.singleton $ buildCwCmd
-            $ set cbSigners [mkSigner' sender00 []]
-            $ set cbChainId (_blockChainId bh)
-            $ set cbCreationTime (toTxCreationTime $ _bct $ _blockCreationTime bh)
-            $ mkCmd (sshow bh) -- nonce is block height, sufficiently unique
-            $ mkExec' "(chain-data)"
+          fmap V.singleton $ buildBasicCmd bh $ mkExec' "(chain-data)"
       }
 
-minerKeysetTest :: TestBlockDb -> IO (IORef MemPoolAccess) -> WebPactExecutionService -> IO ()
-minerKeysetTest bdb _mpRefIO pact = do
+minerKeysetTest :: MultichainTest
+minerKeysetTest _mpRefIO = do
 
   -- run past genesis, upgrades
-  forM_ [(1::Int)..24] $ \_i -> runCut' noMiner
+  forM_ [(1::Int)..24] $ \_i -> runCut'
```

Comment:
I do think Jose's hiding of this indexing is cheap and easy, and more legible.

Reply:
? no indexes are used here

Reply:
Assuming you mean tupling tx builders with tests that could be elegant. The base case is very common so that's probably worth having an abstraction. 

However it's not always so straightforward since some tests are dispatching to different chains, or refilling etc, and getting a fixture to handle those seamlessly is pretty hard. I think indexed access is still necessary for certain tests but they should be able to coexist.

Reply:
I do want some kind of mempool abstraction that can make that behavior easier to model. Probably getting away from concrete `MemPoolAccess` implementations would be an improvement.